libs/payments-ui/src/l10n/l10n.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Children, useEffect, useState, ReactNode } from 'react';
+
+import { negotiateLanguages } from '@fluent/langneg';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { ReactLocalization, LocalizationProvider } from '@fluent/react';
+
+const ftl: Record<string, URL> = {
+  'en-US': new URL('./en-US.ftl', import.meta.url),
+  fr: new URL('./fr.ftl', import.meta.url),
+};
+
+const DEFAULT_LOCALE = 'en-US';
+const AVAILABLE_LOCALES = {
+  'en-US': 'English',
+  fr: 'French',
+};
+
+async function fetchMessages(locale: string): Promise<[string, string]> {
+  const response = await fetch(String(ftl[locale]));
+  const messages = await response.text();
+  return [locale, messages];
+}
+
+function* lazilyParsedBundles(fetchedMessages: Array<[string, string]>) {
+  for (const [locale, messages] of fetchedMessages) {
+    const resource = new FluentResource(messages);
+    const bundle = new FluentBundle(locale);
+    bundle.addResource(resource);
+    yield bundle;
+  }
+}
+
+interface AppLocalizationProviderProps {
+  children: ReactNode;
+}
```

Comment:
nit: not sure if we have a style guide for this, but&hellip; should we define interfaces+types near the top of the file?
I'm guessing they aren't hoisted like functions, but felt a bit weird to be sandwiched between two function definitions. 🥪 
Although I get why maybe it makes sense to keep the interface definition closer to the implementation (L37). 🤷